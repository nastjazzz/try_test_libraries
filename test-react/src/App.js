import './App.css';

function App() {
    return (
        <div className='App'>
            <h1>Hello world!</h1>
            <button>click me</button>
            <input type='text' placeholder='input value.....' />
        </div>
    );
}

export default App;
