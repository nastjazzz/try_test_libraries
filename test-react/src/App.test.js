import { render, screen } from '@testing-library/react';
import App from './App';

describe('app test', () => {
    test('check elems', () => {
        render(<App />);
        const helloWorldElem = screen.getByText(/hello world/i);
        const btnElem = screen.getByRole('button');
        const inputElem = screen.getByPlaceholderText(/input value/i);

        expect(helloWorldElem).toBeInTheDocument();
        expect(btnElem).toBeInTheDocument();
        expect(inputElem).toBeInTheDocument();
        expect(inputElem).toMatchSnapshot();
    });
});
