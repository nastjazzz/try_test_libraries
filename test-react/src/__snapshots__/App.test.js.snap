// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app test check elems 1`] = `
<input
  placeholder="input value....."
  type="text"
/>
`;
